File: IdCaptureSurveyJava/app/src/main/java/com/scandit/datacapture/idcapturesurvey/scan/ScanFragment.java
package com.scandit.datacapture.idcapturesurvey.scan;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.scandit.datacapture.idcapturesurvey.navigation.NavigationUtils;
import com.scandit.datacapture.idcapturesurvey.result.ResultFragment;
import com.scandit.datacapture.idcapturesurveyjava.databinding.FragmentScanBinding;


public class ScanFragment extends Fragment {

    public static String TAG = "scan_fragment";

    @NonNull
    public static ScanFragment create() {
        return new ScanFragment();
    }

    private FragmentScanBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentScanBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setUpIdCapture();
    }

    @Override
    public void onResume() {
        super.onResume();

        // TODO: Switch camera on to start streaming frames
        // TODO: Enable IdCapture
    }

    @Override
    public void onPause() {
        super.onPause();

        // TODO: Switch camera off to stop streaming frames
        // TODO: Disable IdCapture
    }

    private void setUpIdCapture() {
        // TODO: Initialize Scandit SDK and ID Capture
    }

    private void navigateToResult() {
        NavigationUtils.navigateToFragment(requireActivity(), ResultFragment.create(),
                ResultFragment.TAG);
    }
}

File: IdCaptureSurveyKotlin/app/src/main/java/com/scandit/datacapture/idcapturesurvey/scan/ScanFragment.kt
package com.scandit.datacapture.idcapturesurvey.scan

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.scandit.datacapture.idcapturesurvey.databinding.FragmentScanBinding
import com.scandit.datacapture.idcapturesurvey.navigation.navigateToFragment
import com.scandit.datacapture.idcapturesurvey.result.ResultFragment

class ScanFragment private constructor(): Fragment() {

    private lateinit var binding: FragmentScanBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?,
    ): View = FragmentScanBinding.inflate(inflater, container, false)
        .also { binding = it }
        .root

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setUpIdCapture()
    }

    override fun onResume() {
        super.onResume()

        // TODO: Switch camera on to start streaming frames
        // TODO: Enable IdCapture
    }

    override fun onPause() {
        super.onPause()

        // TODO: Switch camera off to stop streaming frames
        // TODO: Disable IdCapture
    }

    private fun setUpIdCapture() {
        // TODO: Initialize Scandit SDK and ID Capture
    }

    private fun navigateToResult() {
        requireActivity().navigateToFragment(ResultFragment.create(), ResultFragment.TAG)
    }

    companion object {
        const val TAG = "scan_fragment"

        fun create() = ScanFragment()
    }
}